drivers/eeprom/eeprom_at2x.c

```diff
@@ -607,6 +625,10 @@ static const struct eeprom_driver_api eeprom_at2x_api = {
 				 SPI_OP_MODE_MASTER | SPI_TRANSFER_MSB | \
 				 SPI_WORD_SET(8), 0) }
 
+#define EEPROM_AT2X_WP_GPIOS(id)					\
+	IF_ENABLED(DT_NODE_HAS_PROP(id, wp_gpios),			\
+		   (.wp_gpio = GPIO_DT_SPEC_GET_OR(id, wp_gpios, {0}),))
```

Comment:
@mbolivar-nordic That's not true. There may be multiple AT2x EEPROMs in the devicetree; some with `wp_gpios`, some without. If one of these instances has `wp_gpios`, the code will be enabled for all - but not present for all. The `GPIO_DT_SPEC_GET_OR()` is correct.

Reply:
Yes, but `.wp_gpio = GPIO_DT_SPEC_GET_OR(id, wp_gpios, {0}),` is inside `IF_ENABLED(DT_NODE_HAS_PROP(id, wp_gpios)`, so the `wp_gpios` property is always defined if the `wp_gpio` variable is initialized, so there is no need of default value. Unfortunately, I can't use `IF_ENABLED(ANY_INST_HAS_WP_GPIOS`, because nested loops are not supported.